util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/mapper/StubMapperUtils.java

```diff
@@ -20,4 +31,28 @@ public static DocumentReferenceProvider referenceProvider(String identifier) {
 	public static DocumentReferenceProvider referenceProvider(String identifier, String routingKey) {
 		return new StubDocumentReferenceProvider( identifier, routingKey );
 	}
+
+	public static <R, O> void expectLoad(ObjectLoader<R, O> objectLoaderMock,
+			Consumer<LoadingDefinitionContext<R, O>> loadingDefinition) {
```

Comment:
Maybe you can use a `Function<Void, Map<R, O>>` instead

Reply:
I don't understand why? `Supplier<Map<R, O>>` could be an idea, but even then I would lose the fluent syntax (`.load(...).load(...)`)

Reply:
Yes a `Supplier<Map<R, O>>`, as wish it not critical at all. Code would look like:

```
StubMapperUtils.expectLoad(
   objectLoaderMock,
   () -> {
     HashMap<StubTransformedReference, StubLoadedObject> result = new HashMap<>();
     result.put( mainTransformedReference, mainLoadedObject );
     result.put( emptyTransformedReference, emptyLoadedObject );
     return result;
   }
 );
```
```
public static <R, O> void expectLoad(ObjectLoader<R, O> objectLoaderMock, Supplier<Map<R, O>> loadingDefinition) {
  EasyMock.expect( objectLoaderMock.load(
    EasyMockUtils.collectionAnyOrderMatcher( new ArrayList<>( loadingDefinition.get().keySet() ) )
  ) )
  .andAnswer(
    () -> ( (List<R>) EasyMock.getCurrentArguments()[0] ).stream()
      .map( loadingDefinition.get()::get )
      .collect( Collectors.toList() )
    );
}
``